### PR TITLE
Cluster-autoscaler: remove nodes that have been unregistered for a long time

### PR DESCRIPTION
**Pull request by mwielgus**
Ref: #2228

cc: @jszczepkowski @fgrzadkowski

### CODE REVIEW

File: cluster-autoscaler/cluster_autoscaler.go [hunk 1]
```diff
@@ -273,6 +275,24 @@ func run(_ <-chan struct{}) {
...
 				}
 
+				// Check if there are any nodes that failed to register in kuberentes
+				// master.
+				unregisteredNodes := autoscalingContext.ClusterStateRegistry.GetUnregisteredNodes()
+				if len(unregisteredNodes) > 0 {
+					glog.V(1).Infof("%d unregistered nodes present", len(unregisteredNodes))
+					removedAny, err := removeOldUnregisteredNodes(unregisteredNodes, &autoscalingContext, time.Now())
+					// There was a problem with removing unregistered nodes. Retry in the next loop.
+					if err != nil {
```
Comment (jszczepkowski) [inline cluster-autoscaler/cluster_autoscaler.go]: You are not handling correctly the case when some nodes were removed, but error was observed afterwards:  "Some unregistered nodes were removed" should be printed before printing error.

File: cluster-autoscaler/utils_test.go [hunk 2]
```diff
@@ -51,3 +55,46 @@ func TestFilterOutSchedulable(t *testing.T) {
...
+	context := &AutoscalingContext{
+		CloudProvider:               provider,
+		ClusterStateRegistry:        clusterState,
+		UnregisteredNodeRemovalTime: 45 * time.Minute,
+	}
+	unregisteredNodes := clusterState.GetUnregisteredNodes()
+	assert.Equal(t, 1, len(unregisteredNodes))
+
+	// Nothing should be removed. The unregistered node is not old enough.
+	removed, err := removeOldUnregisteredNodes(unregisteredNodes, context, time.Now().Add(-50*time.Minute))
```
Comment (jszczepkowski) [inline cluster-autoscaler/utils_test.go]: Nit: do not call Now() multiple times. Call it once and remember the value.

**Comment by jszczepkowski** [comment]
Just two minor comment, after applying them: LGTM

File: cluster-autoscaler/cluster_autoscaler.go (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/cluster_autoscaler.go]: Do you mean the error itself message. Yeah - we can have 2 different here.

Comment (mwielgus) [inline cluster-autoscaler/cluster_autoscaler.go]: Done.

**Pull request merged**